torch/_subclasses/fake_tensor.py

```diff
@@ -931,15 +932,17 @@ class TensorMetadata:
     dense_dim: Optional[int]
     sparse_dim: Optional[int]
 
-    def flatten_into(self, result: List[object]) -> None:
+    def flatten_into(self, result: List[object], state: _CacheKeyState) -> None:
         result.append(self.dtype)
-        result.append(self.shape)
-        result.append(self.stride)
+        for v in self.shape:
+            state.convert_input(result, v)
+        for v in self.stride:
+            state.convert_input(result, v)
         result.append(self.device)
         result.append(self.layout)
         result.append(self.memory_format)
-        result.append(self.storage_offset)
-        result.append(self.storage_bytes)
+        state.convert_input(result, self.storage_offset)
```

Comment:
so i guess the idea is we use `convert_input` for anything that could be py sym type ? add comment /docstring to flatten_into?